### ISSUE TITLE
normalize/denormalize_pixel_coordinates: degenerate height/width/depth silently produce a 2e8 blow-up (size 1) and accept 0 and negative sizes

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3940]
The four public pixel-coordinate normalizers in `kornia/geometry/conversions.py` accept degenerate `height` / `width` / `depth` values silently and return garbage instead of raising. The size is turned into a scale factor by

```python
factor: torch.Tensor = torch.tensor(2.0) / (hw - 1).clamp(eps)
```

(`conversions.py:929` for `normalize_pixel_coordinates`, and `:965`, `:997`, `:1029` for `denormalize_pixel_coordinates`, `normalize_pixel_coordinates3d`, `denormalize_pixel_coordinates3d`). With a size of `1` the denominator is `0`, which `.clamp(eps)` turns into `1e-8` — so instead of an error the user gets a scale factor of `2e8`. There is no validation of `height` / `width` / `depth` anywhere in these functions, so `0` and even negative sizes are accepted too. No `inf`, no `nan`, no warning — just a silently wrong number that will propagate into whatever consumes the coordinates.

## Repro

```python
import torch
from kornia.geometry.conversions import (
    normalize_pixel_coordinates,
    denormalize_pixel_coordinates,
    normalize_pixel_coordinates3d,
)

f64 = torch.float64
print("H=W=1        :", normalize_pixel_coordinates(torch.tensor([[1., 1.]], dtype=f64), 1, 1).tolist())
print("width=0      :", normalize_pixel_coordinates(torch.tensor([[0., 0.]], dtype=f64), 4, 0).tolist())
print("width=-3     :", normalize_pixel_coordinates(torch.tensor([[1., 1.]], dtype=f64), 4, -3).tolist())
print("denorm W=1   :", denormalize_pixel_coordinates(torch.tensor([[0., 0.]], dtype=f64), 4, 1).tolist())
print("3d D=1       :", normalize_pixel_coordinates3d(torch.tensor([[1., 0., 0.]], dtype=f64), 1, 5, 9).tolist())

p = torch.tensor([[1., 2.]], dtype=f64)
print("roundtrip H=5,W=7:", denormalize_pixel_coordinates(normalize_pixel_coordinates(p, 5, 7), 5, 7).tolist())
print("roundtrip H=4,W=1:", denormalize_pixel_coordinates(normalize_pixel_coordinates(p, 4, 1), 4, 1).tolist())
```

## Observed

```
H=W=1        : [[199999999.0, 199999999.0]]
width=0      : [[-1.0, -1.0]]
width=-3     : [[199999999.0, -0.33333333333333337]]
denorm W=1   : [[5e-09, 1.5]]
3d D=1       : [[199999999.0, -1.0, -1.0]]
roundtrip H=5,W=7: [[0.9999999999999998, 2.0]]
roundtrip H=4,W=1: [[1.0, 2.0]]
```

## Expected

- **`H = W = 1`** should either raise or return a defined value. A 1-pixel axis has no corner-aligned normalization — the mapping `2x/(W−1) − 1` is undefined at `W = 1` — so `199999999.0` is not an approximation of anything. A `ValueError` naming the offending dimension is the honest answer; returning `0.0` (mapping the single pixel to the centre) would be a defensible alternative, but it must be a decision, not a clamp artifact.
- **`width = 0`** and **`width = -3`** should raise. A zero or negative image dimension is nonsense on its face. Today `width=0` yields `(0−1) = −1`, which `.clamp(eps)` silently flips *up* to `1e-8`; note this makes the `width=0` output (`[[-1.0, -1.0]]` at `x=0`) look perfectly plausible, so the corruption is invisible at exactly the input a smoke test would use. `width=-3` is accepted with the same `2e8` factor.
- **`normalize` and `denormalize` should remain mutual inverses**, as they are in the non-degenerate case (`H=5, W=7` round-trips to ~2e-16 in float64, shown above). In the degenerate case they fail in *opposite* directions — `normalize` explodes by `2e8` while `denormalize` collapses by `5e-9` (factor `(W−1).clamp(eps)/2`) — so the two errors happen to cancel in a round-trip (`H=4, W=1` returns `[[1.0, 2.0]]`). That cancellation is a coincidence of the two reciprocal clamps, not correctness: either function used on its own returns a wrong number.
- The **3D variants are affected identically** — `normalize_pixel_coordinates3d` with `D = 1` puts `199999999.0` in the depth slot (note the `(d, x, y)` component order).

Suggested fix: validate `height` / `width` / `depth` up front and raise on values `< 1`, then decide explicitly what a size of exactly `1` means and implement that branch rather than relying on `.clamp(eps)`. Whatever is chosen should be applied consistently across all four functions.

## Relationship to #3929

**Different function, different code path, different symptom — this is not a duplicate of #3929, and fixing that issue would not fix this one.**

| | #3929 | this issue |
|---|---|---|
| Function | `crop_by_transform_mat` / `warp_perspective`, via `normal_transform_pixel` | `normalize_pixel_coordinates` and its 3 siblings |
| Code path | `conversions.py:1113-1114`, explicit branch `width_denom = eps if width == 1 else width - 1.0` | `conversions.py:929` (and `:965`, `:997`, `:1029`), `(hw - 1).clamp(eps)` |
| `eps` | `1e-14` → scale `2e14` | `1e-8` → scale `2e8` |
| Symptom | all-`NaN` output, plus a silently dropped projective row on the `align_corners=False` path | **no NaN** — a finite, plausible-looking but wrong number |
| Degenerate sizes reached | `1` | `1`, `0`, and negatives |

They are the same *class* of defect (degenerate size-1 axis under the corner-aligned normalization) and the degenerate-size policy is probably worth settling once for both call sites, but they are independently reachable through different public APIs and need separate fixes.

## Classification

**Clearly-broken-output bug**, not a semantic/default-convention change. `2e8` is not an alternative convention for normalizing a 1-pixel axis; it is a clamp constant leaking into the result. Accepting `width = 0` and negative sizes without validation is likewise not a convention. Per the reference-core lifecycle (Part I), this class may land early as an ordinary bugfix rather than waiting for the coordinated Repair & Deprecation window.

One caveat for the fix PR: if the chosen fix *raises* on size `< 2`, that converts currently-silent calls into errors, which is user-visible for anyone feeding 1-pixel axes through these functions today (their results are wrong, but they are not currently crashing). Worth a scan of internal callers before landing.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
